### DISCUSSION TITLE
How to prevent monitor from turning off on restart?

### DISCUSSION THREAD

**Discussion by netjack7** [JPersson77/LGTVCompanion#238] [Q&A]
Regardless of which checkbox I select restart, shut down, power off, monitor shuts off on restart. How do I prevent that?

**Comment by JPersson77** [comment]
Enable debug log level logging and paste a log here please and I will  have a look

**Reply by netjack7** [reply]
When I tick the checkbox "perform logging", hit restart, and open up the options box, the perform logging is unchecked and clicking on showing does nothing. What am I doing wrong to get you the log?

**Comment by JPersson77** [comment]
Hi, just to confirm did you also click apply?

**Reply by netjack7** [reply]
yup you were right didnt hit apply!

Wed 07:04:05 > LGTV Companion Service started (v 3.1.9) ---------------------------
Wed 07:04:05 > Data path: C:\ProgramData\LGTV Companion\
Wed 07:04:05 > Device1, LG, with IP 192.168.7.25 initiated (Enabled:yes, NewConn:yes, WOL:2, SubnetMask:255.255.255.0, PairingKey:21e67b0bd402a90f08cb15084bc9f0d6, MAC: 4C:BA:D7:E8:32:C4 , VerifyHdmiInput:off, SetHdmiInput:off, BlankOnIdle:off)
Wed 07:04:05 > Host IP detected: 192.168.7.169/24
Wed 07:04:05 > Shutdown timing: default
Wed 07:04:05 > ** System requests displays ON.
Wed 07:04:05 > Device1, spawning Thread_DisplayOn().
Wed 07:04:05 > Device1, repeating WOL broadcast started to MAC: 4CBAD7E832C4 using IP address: 192.168.7.25
Wed 07:04:05 > Device1, Best route to IP - interface index 13 LUID 1689399632855040 route protocol 2
Wed 07:04:06 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"21e67b0bd402a90f08cb15084bc9f0d6"}}
Wed 07:04:06 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":1,"error":"500 Application error","payload":{"returnValue":false,"state":"Active","errorCode":"-102","errorText":"The current sub state must be 'screen off'"}}
Wed 07:04:06 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":2,"payload":{"state":"Active","returnValue":true}}
Wed 07:04:06 > Device1, power state is: ON
Wed 07:04:06 > -daemon 2 started
Wed 07:04:06 > [IPC Daemon 2] Started.
Wed 07:04:06 > Device1, repeating WOL broadcast ended
Wed 07:04:18 > WARNING! The application did not receive an Event Subscription Callback prior to system shutting down. Unable to determine if system is shutting down or restarting.
Wed 07:04:18 > Device1, spawning Thread_DisplayOff().
Wed 07:04:18 > Device1, [DEBUG] (SSL) OFF response 1: {"type":"registered","id":"register_0","payload":{"client-key":"21e67b0bd402a90f08cb15084bc9f0d6"}}
Wed 07:04:18 > Device1, [DEBUG] (SSL) OFF response 2: {"type":"response","id":4,"payload":{"state":"Active","returnValue":true}}
Wed 07:04:19 > Device1, [DEBUG] (SSL) OFF response 4: {"type":"response","id":5,"payload":{"returnValue":true}}
Wed 07:04:19 > Device1, power state is: OFF.
Wed 07:04:20 > The service terminated.

Wed 07:05:12 > LGTV Companion Service started (v 3.1.9) ---------------------------
Wed 07:05:12 > Data path: C:\ProgramData\LGTV Companion\
Wed 07:05:12 > Device1, LG, with IP 192.168.7.25 initiated (Enabled:yes, NewConn:yes, WOL:2, SubnetMask:255.255.255.0, PairingKey:21e67b0bd402a90f08cb15084bc9f0d6, MAC: 4C:BA:D7:E8:32:C4 , VerifyHdmiInput:off, SetHdmiInput:off, BlankOnIdle:off)
Wed 07:05:12 > Host IP detected: 192.168.7.169/24
Wed 07:05:12 > Shutdown timing: default
Wed 07:05:12 > ** System requests displays ON.
Wed 07:05:12 > Device1, spawning Thread_DisplayOn().
Wed 07:05:12 > Device1, repeating WOL broadcast started to MAC: 4CBAD7E832C4 using IP address: 192.168.7.25
Wed 07:05:12 > Device1, Best route to IP - interface index 13 LUID 1689399632855040 route protocol 2
Wed 07:05:12 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"21e67b0bd402a90f08cb15084bc9f0d6"}}
Wed 07:05:12 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":1,"error":"500 Application error","payload":{"returnValue":false,"state":"Screen On","errorCode":"-102","errorText":"The current sub state must be 'screen off'"}}
Wed 07:05:12 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":2,"payload":{"state":"Active Standby","processing":"Screen On","returnValue":true}}
Wed 07:05:13 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"21e67b0bd402a90f08cb15084bc9f0d6"}}
Wed 07:05:13 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":3,"error":"500 Application error","payload":{"returnValue":false,"state":"Screen On","errorCode":"-102","errorText":"The current sub state must be 'screen off'"}}
Wed 07:05:13 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":4,"payload":{"state":"Active Standby","processing":"Screen On","returnValue":true}}
Wed 07:05:14 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"21e67b0bd402a90f08cb15084bc9f0d6"}}
Wed 07:05:14 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":5,"error":"500 Application error","payload":{"returnValue":false,"state":"Screen On","errorCode":"-102","errorText":"The current sub state must be 'screen off'"}}
Wed 07:05:14 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":6,"payload":{"state":"Active Standby","processing":"Screen On","returnValue":true}}
Wed 07:05:16 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"21e67b0bd402a90f08cb15084bc9f0d6"}}
Wed 07:05:16 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":7,"error":"500 Application error","payload":{"returnValue":false,"state":"Active","errorCode":"-102","errorText":"The current sub state must be 'screen off'"}}
Wed 07:05:16 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":8,"payload":{"state":"Active","returnValue":true}}
Wed 07:05:16 > Device1, power state is: ON
Wed 07:05:17 > Device1, repeating WOL broadcast ended
Wed 07:05:30 > -daemon 1 started
Wed 07:05:30 > [IPC Daemon 1] Started.

**Comment by JPersson77** [comment]
The reason is because the syatem did not write the necessary event in the event log. See the corresponding warning message

`Wed 07:04:18 > WARNING! The application did not receive an Event Subscription Callback prior to system shutting down. Unable to determine if system is shutting down or restarting.`

Please install the latest version and try to set the shutdown timing to 'delayed' (if it does  not help to just upgrade as there was many changes since the version you are using).

Please let me know how it goes

**Comment by netjack7** [comment]
Wow what an easy fix - I just upgraded. I thought I was on the latest!
